libraries/windows_updates.rb

```diff
@@ -170,8 +170,18 @@ def fetch_updates
     EOH
     cmd = @inspec.powershell(script)
 
+    result = cmd.stdout
     begin
-      @cache_available = JSON.parse(cmd.stdout)
+      # PowerShell's `ConvertTo-Json` returns an Array of Hashes only if there
+      # is more than one object passed into it, otherwise it returns a single
+      # Hash. This ensures that an Array is always returned
+      @cache_available = if result.is_a?(Array)
+                           outputs = []
+                           result.each { |r| outputs.push(JSON.parse(r)) }
+                           outputs
+                         else
+                           [JSON.parse(result)]
+                         end
```

Comment:
I'd suggest separating normalization from parsing. (And as always just mapping over the array to produce the array you want, instead of making some temporary local variable, and pushing the rescue to the end of your method.)

```ruby
def fetch_updates
  # ...

  # PowerShell's `ConvertTo-Json` returns an Array of Hashes only if there
  # is more than one object passed into it, otherwise it returns a single
  # Hash. This ensures that an Array is always returned
  result = cmd.stdout.is_a?(Hash) ? [result] : result

  @cache_available = result.collect { |r| JSON.parse(r) }
rescue JSON::ParserError => _
  @cache_available = {}
end
```

Reply:
I like this code style @TrevorBramble . @jerryaldrichiii would you like to include this change? Once in, I am happy to do a release of this baseline.

Reply:
I also like this style. Thanks @Trev